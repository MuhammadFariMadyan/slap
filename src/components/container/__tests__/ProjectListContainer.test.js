import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import "jest-styled-components";

import Connected, { ProjectListContainer } from "../ProjectListContainer";

describe("ProjectListContainer", () => {
  jest.mock("../../presentational/ProjectList", () => "ProjectList");

  Date.now = jest.fn(() => 1537028524421);

  const mockStore = configureMockStore();

  const projects = [
    {
      id: "project-1",
      name: "Presentation 1st",
      timestamp: Date.now()
    },
    {
      id: "project-2",
      name: "Presentation 2nd",
      timestamp: Date.now()
    }
  ];

  const slides = [
    {
      id: "slide-1",
      projectID: "project-1"
    },
    {
      id: "slide-2",
      projectID: "project-2"
    }
  ];

  const contents = [
    {
      id: "content-1",
      slideID: "slide-1",
      projectID: "project-1",
      type: "text",
      content: "Hello World",
      position: {
        x: 32,
        y: 32
      }
    },
    {
      id: "content-2",
      slideID: "slide-2",
      projectID: "project-2",
      type: "text",
      content: "Hello Code",
      position: {
        x: 64,
        y: 64
      }
    }
  ];

  const setup = propOverrides => {
    const props = {
      items: [...Array(3)].map((_, index) => ({
        id: `project-${index + 1}`,
        name: `Project ${index + 1}`,
        timestamp: 1537004755445,
        contents: [
          {
            id: "content-1",
            slideID: "slide-1",
            projectID: "project-1",
            type: "text",
            content: "Hello World",
            position: {
              x: 32,
              y: 32
            }
          },
          {
            id: "content-2",
            slideID: "slide-1",
            projectID: "project-1",
            type: "text",
            content: "Stay hungry",
            position: {
              x: 128,
              y: 96
            }
          }
        ]
      })),
      onClick: jest.fn(),
      onClickDelete: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<ProjectListContainer {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const store = mockStore({
      projects,
      slides,
      contents
    });

    const wrapper = shallow(<Connected store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders as connected", () => {
    const { wrapper } = setupConnected();

    expect(wrapper).toMatchSnapshot();
  });

  it("should map new to props", () => {
    const { wrapper } = setupConnected();

    expect(wrapper.props()).toMatchObject({
      items: expect.any(Array),
      onClick: expect.any(Function),
      onClickDelete: expect.any(Function)
    });
  });

  it("should calls 'onClick' props", () => {
    const { wrapper, store } = setupConnected();

    wrapper.props().onClick(projects[0]);

    expect(store.getActions()).toEqual([
      {
        type: "SET_ACTIVE_PROJECT_ID",
        payload: projects[0].id
      },
      {
        type: "SET_ACTIVE_SLIDE_ID",
        payload: slides[0].id
      }
    ]);
  });

  it("should calls 'onClickDelete' props", () => {
    const { wrapper, store } = setupConnected();

    wrapper.props().onClickDelete(projects[0]);

    expect(store.getActions()).toEqual([
      {
        type: "SET_ACTIVE_PROJECT_ID",
        payload: null
      },
      {
        type: "SET_ACTIVE_SLIDE_ID",
        payload: null
      },
      {
        type: "REMOVE_PROJECT",
        payload: projects[0].id
      },
      {
        type: "REMOVE_SLIDE",
        payload: slides[0].id
      },
      {
        type: "REMOVE_CONTENT",
        payload: contents[0].id
      }
    ]);
  });
});
