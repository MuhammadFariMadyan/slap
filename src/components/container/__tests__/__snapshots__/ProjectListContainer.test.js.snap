// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListContainer should renders as connected 1`] = `
<Component
  deleteProject={[Function]}
  items={
    Array [
      Object {
        "contents": Array [
          Object {
            "content": "Hello World",
            "id": "content-1",
            "position": Object {
              "x": 32,
              "y": 32,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
        ],
        "id": "project-1",
        "name": "Presentation 1st",
        "timestamp": 1537028524421,
      },
      Object {
        "contents": Array [
          Object {
            "content": "Hello Code",
            "id": "content-2",
            "position": Object {
              "x": 64,
              "y": 64,
            },
            "projectID": "project-2",
            "slideID": "slide-2",
            "type": "text",
          },
        ],
        "id": "project-2",
        "name": "Presentation 2nd",
        "timestamp": 1537028524421,
      },
    ]
  }
  onClick={[Function]}
  onClickDelete={[Function]}
  openProject={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`ProjectListContainer should renders default 1`] = `
<ProjectList
  items={
    Array [
      Object {
        "contents": Array [
          Object {
            "content": "Hello World",
            "id": "content-1",
            "position": Object {
              "x": 32,
              "y": 32,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
          Object {
            "content": "Stay hungry",
            "id": "content-2",
            "position": Object {
              "x": 128,
              "y": 96,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
        ],
        "id": "project-1",
        "name": "Project 1",
        "timestamp": 1537004755445,
      },
      Object {
        "contents": Array [
          Object {
            "content": "Hello World",
            "id": "content-1",
            "position": Object {
              "x": 32,
              "y": 32,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
          Object {
            "content": "Stay hungry",
            "id": "content-2",
            "position": Object {
              "x": 128,
              "y": 96,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
        ],
        "id": "project-2",
        "name": "Project 2",
        "timestamp": 1537004755445,
      },
      Object {
        "contents": Array [
          Object {
            "content": "Hello World",
            "id": "content-1",
            "position": Object {
              "x": 32,
              "y": 32,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
          Object {
            "content": "Stay hungry",
            "id": "content-2",
            "position": Object {
              "x": 128,
              "y": 96,
            },
            "projectID": "project-1",
            "slideID": "slide-1",
            "type": "text",
          },
        ],
        "id": "project-3",
        "name": "Project 3",
        "timestamp": 1537004755445,
      },
    ]
  }
  onClick={[Function]}
  onClickDelete={[Function]}
/>
`;
